thrust/thrust/detail/complex/arithmetic.h

```diff
@@ -1,303 +0,0 @@
-/*
- *  Copyright 2008-2021 NVIDIA Corporation
- *  Copyright 2013 Filipe RNC Maia
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- */
-
-#pragma once
-
-#include <thrust/detail/config.h>
-
-#include <thrust/complex.h>
-#include <thrust/detail/complex/c99math.h>
-#include <cfloat>
-#include <cmath>
-
-THRUST_NAMESPACE_BEGIN
-
-  /* --- Binary Arithmetic Operators --- */
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator+(const complex<T0>& x, const complex<T1>& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x.real() + y.real(), x.imag() + y.imag());
-}
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator+(const complex<T0>& x, const T1& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x.real() + y, x.imag());
-}
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator+(const T0& x, const complex<T1>& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x + y.real(), y.imag());
-}
-
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator-(const complex<T0>& x, const complex<T1>& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x.real() - y.real(), x.imag() - y.imag());
-}
```

Comment:
The following code used to work, but now it doesn't:
```cpp
  thrust::complex<float> a;
  thrust::complex<double> b;
  auto c = a + b;
```
I think we need to keep this overload and maybe mark it as deprecated. Same with other operators.